### DISCUSSION TITLE
Visualise chapter through API key

### DISCUSSION THREAD

**Discussion by verbuyst** [readium/kotlin-toolkit#573] [Ideas]
Hi everybody,

I use ReadEra the most for my Epub files, but as a visual person I would like to have an Ai visualise the chapter I'm reading before I start reading that chapter. Now nobody has this feature in any app in the Play store and I have to do it manually: 

Step one: Open the Epub file and copy/paste the chapter into chatgpt
Step two: Chatgpt sumerise the chapter into 2000 characters (this is the max Flux accepts)
Step three: Aitubo uses Flux Ai to create an image of that summery of chatgpt
Step four: Add the pictures into the Epub before the chapters
Step Five: Reading the Epub on any app

I saw this open source app and tried it myself in Android Studio (with the help of Chatgpt), but for the love of god I can't get it to work...
I get errors all over the place when I "clean and build" the app to test it through the emulator and apk... I have tried the Test-app source code and the full project, every time I get errors :-(

Chatgpt say's to try one thing, that breaks other stuff inside the code and I get the feeling I'm running in circles, I have my own API key's so any user could just copy/paste his api key's inside the settings menu and activate this feature if he/she wants this.

If anybody could get this to work it would be an excellent feature :-)

**Comment by qnga** [comment]
Hello,
There's no error for other people, so we can't help if you don't tell which kind of errors you get.

**Comment by HadrienGardeur** [comment]
> Step one: Open the Epub file and copy/paste the chapter into chatgpt
Step two: Chatgpt sumerise the chapter into 2000 characters (this is the max Flux accepts)
Step three: Aitubo uses Flux Ai to create an image of that summery of chatgpt
Step four: Add the pictures into the Epub before the chapters
Step Five: Reading the Epub on any app

I'm pretty sure that all of this could be done using Gemini Nano in the future, which would open the door to on-device/offline generation of such summary/visualization.

For now this is limited to newer Pixel/Samsung devices, but in two to three years time, this should be generally available on most devices that people own.
